### PR TITLE
Implement x-ms-parameterized-host extension for ruby

### PR DESCRIPTION
**Pull request by vishrutshah**
This is regarding Issue #722. 

Implementing support for x-ms-parameterized-host in Ruby generator.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vishrutshah__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: gulpfile.js [hunk 1]
```diff
@@ -621,8 +623,8 @@ gulp.task('test:nugetPackages:clean', function () {
 });
 
 var autoRestExe = function(){
-  gutil.log(glob.sync(path.join(basePathOrThrow(), 'AutoRest/NugetPackageTest/packages/autorest.*/tools/AutoRest.exe')));
-  return glob.sync(path.join(basePathOrThrow(), 'AutoRest/NugetPackageTest/packages/autorest.*/tools/AutoRest.exe'))[0];
+  gutil.log(glob.sync(path.join(basePathOrThrow(), 'binaries/net45/AutoRest.exe')));
```
Comment (vishrutshah) [inline gulpfile.js]: Needs to be reverted.

File: AutoRest/Generators/Ruby/Ruby/TemplateModels/MethodTemplateModel.cs [hunk 2]
```diff
@@ -30,6 +32,172 @@ public MethodTemplateModel(Method source, ServiceClient serviceClient)
...
+        /// Gets the skip encoding path parameters
+        /// </summary>
+        public virtual IEnumerable<ParameterTemplateModel> SkipEncodingPathParams
+        {
+            get
+            {
+                return AllPathParams.Where(p => 
+                    (p.Extensions.ContainsKey(Generator.Extensions.SkipUrlEncodingExtension) &&
+                    String.Equals(p.Extensions[Generator.Extensions.SkipUrlEncodingExtension].ToString(), "true", StringComparison.OrdinalIgnoreCase) &&
+                    !p.Extensions.ContainsKey("hostParameter")));
```
Comment (veronicagg) [inline AutoRest/Generators/Ruby/Ruby/TemplateModels/MethodTemplateModel.cs]: why do we condition on not containing key "hostParameter" here and not in the "EncodingPathParams" method?

File: AutoRest/Generators/Ruby/Ruby/TemplateModels/MethodTemplateModel.cs [hunk 3]
```diff
@@ -30,6 +32,172 @@ public MethodTemplateModel(Method source, ServiceClient serviceClient)
...
+            }
+        }
+
+        /// <summary>
+        /// Gets the path parameters not including the params that skip encoding
+        /// </summary>
+        public virtual IEnumerable<ParameterTemplateModel> EncodingPathParams
```
Comment (veronicagg) [inline AutoRest/Generators/Ruby/Ruby/TemplateModels/MethodTemplateModel.cs]: should swap order of this method and the one above?

Comment (vishrutshah) [inline AutoRest/Generators/Ruby/Ruby/TemplateModels/MethodTemplateModel.cs]: Yes makes sense.

**Comment by amarzavery** [comment]
@azuresdkci - Test this please.

File: AutoRest/Generators/Ruby/Ruby/TemplateModels/MethodTemplateModel.cs (hunk 2 above)
Comment (vishrutshah) [inline AutoRest/Generators/Ruby/Ruby/TemplateModels/MethodTemplateModel.cs]: That's an excellent point. Let me give a more thought about how they should be...

**Comment by veronicagg** [comment]
Looks good @vishrutshah . Thanks!

**Comment by vishrutshah** [comment]
Thanks for the review !!

**Pull request merged**